datafusion/common/src/lib.rs

```diff
@@ -27,3 +27,23 @@ pub use column::Column;
 pub use dfschema::{DFField, DFSchema, DFSchemaRef, ExprSchema, ToDFSchema};
 pub use error::{field_not_found, DataFusionError, Result, SchemaError};
 pub use scalar::{ScalarType, ScalarValue};
+
+#[macro_export]
+macro_rules! downcast_value {
```

Comment:
It would be great to add some doc comments here (especially as this will now become part of the public API). It could be done as a follow on PR too

Reply:
I took the liberty of adding some brief docs here